control-plane/agents/src/bin/core/volume/operations.rs

```diff
@@ -473,7 +486,10 @@ impl ResourcePublishing for OperationGuardArc<VolumeSpec> {
         }
 
         // Get the newer target node for the new nexus creation.
-        let nexus_node = self.next_target_node(registry, request, true).await?;
+        let nexus_node = match nexus_node {
+            Some(node) => node,
+            None => self.next_target_node(registry, request, true).await?,
```

Comment:
Just wondering, can two times calling `next_target_node` can produce different results(if something moved/changed in background), even if it does should it matter? 

Do we need to recall the `next_target_node` here? Can we simply error out on the `None` arm?

Reply:
hmm good point.
No we can't error out on None, but I've reworked the code allowing us to fail on error, PTAL